### PR TITLE
feat: Add GridFS Basic API

### PR DESCRIPTION
**Pull request by lucsoft**
Closes https://github.com/denodrivers/deno_mongo/issues/238

Tasks:
- [x] Add Interfaces that cover the Basic API
- [x] Delete API (was tested)
- [x] Upload (was tested)
- [x] Download (was tested)
- [x] Find (was tested)
- [x] Add missing Index checks
- [x] Add Tests

### CODE REVIEW

**Comment by lucsoft** [comment]
deno.land has currently some issues and I can't really control that, but the work is done

**Comment by lucsoft** [comment]
@erfanium do you want to review it?

File: src/gridfs/upload.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  options?: GridFSUploadOptions,
+) {
+  const chunkSizeBytesCombined = options?.chunkSizeBytes ?? chunkSizeBytes;
+  const uploadBuffer = new Uint8Array(new ArrayBuffer(chunkSizeBytesCombined));
+  let bufferPosition = 0;
+  let chunksInserted = 0;
+  let fileSizeBytes = 0;
+  return new WritableStream<Uint8Array>({
+    write: async (chunk: Uint8Array) => {
+      const loop = async (chunk: Uint8Array) => {
```
Comment (erfanium) [inline src/gridfs/upload.ts:24]: Maybe another solution (e.g. loops) to avoid recursive function calls?

File: src/types.ts [hunk 2]
```diff
@@ -143,35 +144,6 @@ export interface UpdateOptions {
   // session?: ClientSession
 }
 
-/**
- * interface for WriteConcern documents used by MongoDB
- *
- * @see https://docs.mongodb.com/manual/reference/write-concern/
- */
-export interface WriteConcern {
```
Comment (erfanium) [inline src/types.ts:151]: Why `WriteConcern` interface deleted?

File: tests/cases/06_gridfs.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  testWithClient(
+    "GridFS: Echo large Image (compare with different Image)",
+    async (client) => {
+      const bucket = new GridFSBucket(client.database("test"), {
+        bucketName: "deno_logo",
+      });
+
+      const upstream = bucket.openUploadStream("deno_logo.png");
+
+      await (await fetch("https://deno.land/images/deno_logo.png")).body!
```
Comment (erfanium) [inline tests/cases/06_gridfs.ts:62]: These urls aren't supposed to always be available. better to load files from disk directly. but it's okay for now

File: src/types.ts (hunk 2 above)
Comment (lucsoft) [reply src/types.ts:151]: It's not I just moved it out of the massive types declare file to readwriteconcers that's how the spec is also designed

File: src/gridfs/upload.ts (hunk 1 above)
Comment (lucsoft) [reply src/gridfs/upload.ts:24]: @manuelaguirre do you want to change it to a loop?

Comment (manuelaguirre) [reply src/gridfs/upload.ts:24]: Sure. https://github.com/lucsoft/deno_mongo/pull/3

File: tests/cases/06_gridfs.ts (hunk 3 above)
Comment (manuelaguirre) [reply tests/cases/06_gridfs.ts:62]: We could track this as a chore

**Review by erfanium — APPROVED** [review]
Great!

**Comment by manyuanrong** [comment]
Nice work!

**Review by manyuanrong — APPROVED** [review]

**Pull request merged**